jdaviz/configs/specviz/plugins/parsers.py

```diff
@@ -159,6 +160,15 @@ def specviz_spectrum1d_parser(app, data, data_label=None, format=None, show_in_v
             # Make metadata layout conform with other viz.
             spec.meta = standardize_metadata(spec.meta)
 
+            # If this is the first loaded data, we want to set spectral y unit type to Flux or
+            # Surface Brightness as appropriate
+            if len(app.data_collection) == 0 and "Unit Conversion" in app._jdaviz_helper.plugins:
+                uc = app._jdaviz_helper.plugins["Unit Conversion"]
+                if check_if_unit_is_per_solid_angle(flux_units):
+                    uc._obj.spectral_y_type = "Surface Brightness"
+                else:
+                    uc._obj.spectral_y_type = "Flux"
```

Comment:
I wonder if this logic should be handled across all viewers/configs in a listener in the plugin itself and merged with the eventual general solution to #3168.  But this makes sense as a simple fix until we can do that.